pkg/dockerregistry/server/repositorymiddleware.go

```diff
@@ -473,11 +473,15 @@ func (r *repository) signedManifestFillImageMetadata(manifest *schema1.SignedMan
 		}
 		layer.LayerSize = desc.Size
 		// count empty layer just once (empty layer may actually have non-zero size)
-		if !layerSet.Has(layer.Name) {
+		if !blobSet.Has(layer.Name) {
 			image.DockerImageMetadata.Size += desc.Size
-			layerSet.Insert(layer.Name)
+			blobSet.Insert(layer.Name)
 		}
 	}
+	if len(image.DockerImageConfig) > 0 && !blobSet.Has(image.DockerImageMetadata.ID) {
+		blobSet.Insert(image.DockerImageMetadata.ID)
+		image.DockerImageMetadata.Size += int64(len(image.DockerImageConfig))
```

Comment:
> the size of the ID is not already included?

The ID is a name of the config. It's actually a digest - similar to layer's digests. The size needs to be updated for the size of the config itself, not its ID.

Update of `blobSet` is actually a useless operation here, but I'd keep that for a sake of correctness in case we append another size updating block codes afterwards.

> would like a test around this size calculation

I'm working on tests.